src/megatron/bridge/data/vlm_datasets/hf_provider.py

```diff
@@ -0,0 +1,116 @@
+# Copyright (c) 2025, NVIDIA CORPORATION.  All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""
+Provider that builds conversation datasets from HuggingFace datasets.
+"""
+
+from dataclasses import dataclass
+from typing import Any, Callable, Dict, List, Optional, Tuple
+
+import torch
+from transformers import AutoProcessor
+
+from megatron.bridge.training.config import DatasetBuildContext, DatasetProvider
+
+from .conversation_dataset import VLMConversationDataset
+from .hf_dataset_makers import (
+    make_cord_v2_dataset,
+    make_cv17_dataset,
+    make_medpix_dataset,
+    make_rdr_dataset,
+)
+
+
+@dataclass(kw_only=True)
+class HFDatasetConversationProvider(DatasetProvider):
+    """DatasetProvider that builds VLM conversation datasets from HF datasets.
+
+    This provider leverages simple maker functions that return lists of examples
+    with a "conversation" schema understood by model processors. It binds a
+    HuggingFace `AutoProcessor` for the specified model and selects an
+    appropriate collate function for batching.
+    """
+
+    # Required to match model.seq_length (enforced by ConfigContainer.validate)
+    sequence_length: int
```

Comment:
is it strictly needed? or should we update the validation logic to handle the dataset config not having sequence length set?

Reply:
it's needed for some cases we need to do truncation or padding. maybe we can somehow consolidate those? but feels like a lower priority